libs/posthog_client_lite/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+//! A lite version of the PostHog client that only supports local evaluation of feature flags.
+
+use std::{collections::HashMap, sync::Arc};
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use sha2::Digest;
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationResponse {
+    #[allow(dead_code)]
+    flags: Vec<PostHogLocalEvaluationFlag>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlag {
+    key: String,
+    filters: PostHogLocalEvaluationFlagFilters,
+    active: bool,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilters {
+    groups: Vec<PostHogLocalEvaluationFlagFilterGroup>,
+    multivariate: PostHogLocalEvaluationFlagMultivariate,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilterGroup {
+    variant: Option<String>,
+    properties: Option<Vec<PostHogLocalEvaluationFlagFilterProperty>>,
+    rollout_percentage: f64,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilterProperty {
+    key: String,
+    value: PostHogFlagFilterPropertyValue,
+    operator: String,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum PostHogFlagFilterPropertyValue {
+    String(String),
+    Number(f64),
+    Boolean(bool),
+    List(Vec<String>),
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariate {
+    variants: Vec<PostHogLocalEvaluationFlagMultivariateVariant>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariateVariant {
+    key: String,
+    rollout_percentage: f64,
+}
+
+pub struct PostHogClient {
+    server_api_key: String,
+    client_api_key: String,
+    project_id: String,
+    private_api_url: String,
+    public_api_url: String,
+    client: Arc<reqwest::Client>,
+}
+
+pub struct FeatureStore {
+    flags: HashMap<String, PostHogLocalEvaluationFlag>,
+}
+
+impl Default for FeatureStore {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl FeatureStore {
+    pub fn new() -> Self {
+        Self {
+            flags: HashMap::new(),
+        }
+    }
+
+    pub fn update_flags(&mut self, flags: Vec<PostHogLocalEvaluationFlag>) {
+        self.flags.clear();
+        for flag in flags {
+            if flag.active {
+                self.flags.insert(flag.key.clone(), flag);
+            }
+        }
+    }
+
+    /// Generate a consistent hash for a user ID (e.g., tenant ID).
+    ///
+    /// The implementation is different from PostHog SDK. In PostHog SDK, it is sha1 of `user_id.distinct_id.salt`.
+    /// However, as we do not upload all of our tenant IDs to PostHog, we do not have the PostHog distinct_id for a
+    /// tenant.
+    fn consistent_hash(user_id: &str) -> f64 {
```

Comment:
Now the user hash is the same for all flags, so there will be users who always get new features first, and those who get them the last.

It makes sense to add flag_id/flag_name to the hash, so the roll-out order will be different for different flags